src/main/scala/com.snowplowanalytics/iglu.schemaddl/migrations/Migration.scala

```diff
@@ -120,4 +119,84 @@ object Migration {
       .groupBy(_._1)
       .mapValues(_.map(_._2))
 
+  /**
+    * Build a map of source Schema to its OrderedSubSchemas, where all source Schemas
+    * are last version of their model group
+    * @param schemas source Schemas
+    * @param migrationMap migration map of each Schema to list of all available migrations
```

Comment:
As I said before, we are getting NonEmptyList building error with `buildMigrationMap` function when there is only one schema for a model group (e.g. there is only first version schema of the model group). If we give migrationMap as argument to `buildOrderedSubSchemas` function, we can give empty map as migration map in these kind of situations and we can create ordered subschemas for them too. Otherwise, we should implement new function in order to create ordered subschemas for them.